evhtp.c

```diff
@@ -5274,7 +5330,7 @@ evhtp__new_(evhtp_t ** out, struct event_base * evbase, void * arg)
     htp->arg          = arg;
     htp->evbase       = evbase;
     htp->flags        = EVHTP_FLAG_DEFAULTS;
-    htp->bev_flags    = BEV_OPT_CLOSE_ON_FREE;
+    htp->bev_flags    = BEV_OPT_CLOSE_ON_FREE | BEV_OPT_DEFER_CALLBACKS;
```

Comment:
I say we make DEFER_CALLBACKS default. It just takes a lot of pain away in the long term.